clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -67,16 +71,22 @@ class ApiClient(apiUrl: String, accessKey: String, secretKey: String) {
     def call(): String = fn()
   }
 
-  def getStorageNamespace(repoName: String): String = {
+  def getStorageNamespace(repoName: String, accessType: StorageAccessType): String = {
     storageNamespaceCache.get(
       repoName,
       new CallableFn(() => {
         val repo = repositoriesApi.getRepository(repoName)
 
-        ApiClient
-          .translateURI(URI.create(repo.getStorageNamespace), getBlockstoreType())
-          .normalize()
-          .toString
+        val storageNamespace =  accessType match {
+          case StorageAccessType.HadoopFS  =>
+            ApiClient
+              .translateURI(URI.create(repo.getStorageNamespace), getBlockstoreType())
+              .normalize()
+              .toString
+          case StorageAccessType.DirectAccess  => repo.getStorageNamespace
```

Comment:
I am not sure what the difference is between these two "access types".  Also, "direct access" in `lakectl` is to do essentially what lakeFSFS does, so this name is confusing.

Reply:
The idea is to distinguish between the following cases:
1. Needed a storage namespace in a format that some hadoop file system can work with (e.g. should start with s3a for s3aFileSystem, or be of the form "abfs://${container}@${storageAccountName}.dfs.core.windows.net/${blobPath}" for AzureBlobFileSystem - I call this HadoopFS access type
2. Needed a storage namespace in a format used by SDK clients to access the storage directly - I call this direct access type. 

I'm very open to other naming suggestions :) 

Maybe I can use something related to the type of client that uses it? such as StorageClientType with HadoopFS, SDKclient? WDYT?

Reply:
@arielshaqed WDYT about this?

Reply:
Changed this to StorageClientType: SDKClient, HadoopFS